src/main/java/uk/gov/pay/directdebit/events/model/GovUkPayEvent.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.pay.directdebit.events.model;
+
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.payments.model.Payment;
+
+import java.time.ZonedDateTime;
+
+public class GovUkPayEvent {
+
+    private final Long id;
+    private final Long mandateId;
+    private final Long paymentId;
+    private final ZonedDateTime eventDate;
+    private final ResourceType resourceType;
+    private final String eventType;
+
+    public GovUkPayEvent(Long id, Long mandateId, Long paymentId, ZonedDateTime eventDate, ResourceType resourceType, String eventType) {
+        this.id = id;
+        this.mandateId = mandateId;
+        this.paymentId = paymentId;
+        this.eventDate = eventDate;
+        this.resourceType = resourceType;
+        this.eventType = eventType;
```

Comment:
Can we use `Objects.requireNonNull(…)` for the fields that should always be populated?

Reply:
done